includes/multi-currency/wc-payments-multi-currency.php

```diff
@@ -26,3 +26,15 @@ function WC_Payments_Multi_Currency() { // phpcs:ignore WordPress.NamingConventi
 function wcpay_multi_currency_deactivated() {
 	WCPay\MultiCurrency\MultiCurrency::remove_woo_admin_notes();
 }
+
+/**
+ * Gets the switcher widget markup.
+ *
+ * @param array $instance The widget's instance settings.
+ * @param array $args     The widget's arguments.
+ *
+ * @return string The widget markup.
+ */
+function get_wcpay_multi_currency_widget_markup( array $instance = [], array $args = [] ): string {
```

Comment:
Firstly, the precedent is to prefixed with with the name space and then the function name.

Secondly, since MC is written in a way that makes it easy to move later we should we leave it outside the wcpay namespace? Maybe prefix with `wc`?

Adding the word `widget` also feels limiting. Theme authors may not necessarily use this in a widget area.


I think something like `wc_rendor_mult_currency_selector`/ `wc_rendor_mc_selector` or if we stick with wcpay `wcpay_render_multi_currency_selector`.

Reply:
AFAICS this doesn't render the widget directly, it returns the code it produces (widget markup), so it'll be better like `wcpay_get_multi_currency_widget_markup`. Also, AFAIK "widget" doesn't refer to widget area, each one piece inside the widget area are called "widget"s. So it may be better to keep it to point out that code represents a widget.

Reply:
Taha is correct here, it returns the markup and doesn't directly render the widget itself. I understand how the use of the term "widget" makes it seem limiting. What about `wc_multi_currency_get_currency_switcher_markup`, which puts the full namespace first, then the action of getting the Currency Switcher's markup.

Reply:
Oh, so the API would simply return the markup to theme authors. In that case, the name you suggested Jessse, makes sense.  On another note, I **think** `wc_get_currency_switcher_markup` is much simpler as I don't think we need the multi-currency namespace, we're also already suggesting its related to MC by the name currency switcher. The `wc` namespace gives us enough protection. If you don't think this is a good suggestion I'm happy to go with what you already have.

Reply:
Sweet, I went with the shorter version suggested here: https://github.com/Automattic/woocommerce-payments/pull/2849/commits/aa3599b6aaf53f2d291e09f32aa3f440a49b08e0 . I also added a `function_exists` wrapper to prevent any possible name clashes, though there should never be any.